### ISSUE TITLE
[Android] [Crash] RevenueCat incompatibility

### ISSUE THREAD

**Issue by MinseokKang003** [OneSignal/OneSignal-Flutter-SDK#585]
**Description:**
On Android, OneSignal crashes the app when `setAppId` is called.
Crash only happens when RevenueCat dependency is added.
Since I need both notification and in-app purchase functionalities,
the dependencies should be compatible with each other.

**Environment**
1. OneSignal Flutter SDK Version: 3.4.0
2. RevenueCat SDK Version: 4.0.0-rc.2

**Steps to Reproduce Issue:**
1. Run OneSignal Example
2. Add OneSignal Dependency(4.0.0-rc.2)
3. Project crashes

**Anything else:**
Crash logs:
```
D/OneSignal(31495): App id set for first time:  [APP ID]
D/Kiwi    (31495): AppstoreSDK: Starting initialization process for application: [App Bundle ID]
D/Kiwi    (31495): ResourceManagerImpl: Registering resource: com.amazon.a.a.k.c@cdb992d
E/AndroidRuntime(31495): FATAL EXCEPTION: OS_REST_SUCCESS_CALLBACK
E/AndroidRuntime(31495): Process: [App Bundle Id], PID: 31495
E/AndroidRuntime(31495): com.amazon.a.a.o.a.b: Executing thread must be thread: 2, was: 29839
E/AndroidRuntime(31495): 	at com.amazon.a.a.o.a.a.a(Assert.java:59)
E/AndroidRuntime(31495): 	at com.amazon.a.a.o.a.a.a(Assert.java:48)
E/AndroidRuntime(31495): 	at com.amazon.a.a.o.a.a.a(Assert.java:31)
E/AndroidRuntime(31495): 	at com.amazon.a.a.k.c.a(ResourceManagerImpl.java:50)
E/AndroidRuntime(31495): 	at com.amazon.a.a.k.c.<init>(ResourceManagerImpl.java:34)
E/AndroidRuntime(31495): 	at com.amazon.a.a.b(AppstoreSDK.java:169)
E/AndroidRuntime(31495): 	at com.amazon.a.a.<init>(AppstoreSDK.java:153)
E/AndroidRuntime(31495): 	at com.amazon.a.a.a(AppstoreSDK.java:232)
E/AndroidRuntime(31495): 	at com.amazon.device.iap.PurchasingService.registerListener(PurchasingService.java:89)
E/AndroidRuntime(31495): 	at com.onesignal.TrackAmazonPurchase.setListener(TrackAmazonPurchase.java:107)
E/AndroidRuntime(31495): 	at com.onesignal.TrackAmazonPurchase.<init>(TrackAmazonPurchase.java:84)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignal.handleAmazonPurchase(OneSignal.java:985)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignal.init(OneSignal.java:862)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignal.setAppId(OneSignal.java:735)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignal.reassignDelayedInitParams(OneSignal.java:1181)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignal.onRemoteParamSet(OneSignal.java:889)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignal$7.complete(OneSignal.java:1122)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignalRemoteParams.processJson(OneSignalRemoteParams.java:211)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignalRemoteParams.access$100(OneSignalRemoteParams.java:12)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignalRemoteParams$1.onSuccess(OneSignalRemoteParams.java:156)
E/AndroidRuntime(31495): 	at com.onesignal.OneSignalRestClient$5.run(OneSignalRestClient.java:279)
E/AndroidRuntime(31495): 	at java.lang.Thread.run(Thread.java:919)
```

**Comment by giovanni256** [comment]
Following, same issue here

**Comment by giovanni256** [comment]
Found this issue reported in [RevenueCat](https://github.com/RevenueCat/purchases-flutter/issues/412).

The issue was due to Amazon SDK 3.0.2, so RevenueCat updated it to 3.0.3 but we still need to wait for a newer release of purchases_flutter > 4.0.0

**Comment by nan-li** [comment]
Thank you @MinseokKang003 and @giovanni256 for digging into this. 

Since it doesn't seem like there's anything for OneSignal's SDK to implement, I will close this issue.

Please reach out if you run into any future problems.

**Comment by giovanni256** [comment]
We need to re-open it again.

Without OneSignal initialization, RevenueCat is working without fatal errors.
I attach the stack:

`V/OneSignal( 6899): Starting OneSignal initialization!
V/OneSignal( 6899): No class found, not setting up OSRemoteNotificationReceivedHandler
D/OneSignal( 6899): OneSignal handleActivityLifecycleHandler inForeground: true
D/OneSignal( 6899): Application foregrounded focus time: 1318056
D/Kiwi    ( 6899): AppstoreSDK: Starting initialization process for application: it.iovara.trainassistant
D/Kiwi    ( 6899): ResourceManagerImpl: Registering resource: com.amazon.a.a.k.c@bcced4d
E/AndroidRuntime( 6899): FATAL EXCEPTION: OS_REST_SUCCESS_CALLBACK
E/AndroidRuntime( 6899): Process: it.iovara.trainassistant, PID: 6899
E/AndroidRuntime( 6899): com.amazon.a.a.o.a.b: Executing thread must be thread: 2, was: 647
E/AndroidRuntime( 6899): 	at com.amazon.a.a.o.a.a.a(Assert.java:59)
E//AndroidRuntime( 6899): 	at com.amazon.a.a.o.a.a.a(Assert.java:48)
E/AndroidRuntime( 6899): 	at com.amazon.a.a.o.a.a.a(Assert.java:31)
E/AndroidRuntime( 6899): 	at com.amazon.a.a.k.c.a(ResourceManagerImpl.java:50)
E/AndroidRuntime( 6899): 	at com.amazon.a.a.k.c.<init>(ResourceManagerImpl.java:34)
E/AndroidRuntime( 6899): 	at com.amazon.a.a.b(AppstoreSDK.java:171)
E/AndroidRuntime( 6899): 	at com.amazon.a.a.<init>(AppstoreSDK.java:155)
E/AndroidRuntime( 6899): 	at com.amazon.a.a.a(AppstoreSDK.java:234)
E/AndroidRuntime( 6899): 	at com.amazon.device.iap.PurchasingService.registerListener(PurchasingService.java:89)
E/AndroidRuntime( 6899): 	at com.onesignal.TrackAmazonPurchase.setListener(TrackAmazonPurchase.java:107)
E/AndroidRuntime( 6899): 	at com.onesignal.TrackAmazonPurchase.<init>(TrackAmazonPurchase.java:84)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignal.handleAmazonPurchase(OneSignal.java:985)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignal.init(OneSignal.java:862)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignal.setAppId(OneSignal.java:735)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignal.reassignDelayedInitParams(OneSignal.java:1181)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignal.onRemoteParamSet(OneSignal.java:889)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignal$7.complete(OneSignal.java:1122)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignalRemoteParams.processJson(OneSignalRemoteParams.java:211)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignalRemoteParams.access$100(OneSignalRemoteParams.java:12)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignalRemoteParams$1.onSuccess(OneSignalRemoteParams.java:156)
E/AndroidRuntime( 6899): 	at com.onesignal.OneSignalRestClient$5.run(OneSignalRestClient.java:279)
E/AndroidRuntime( 6899): 	at java.lang.Thread.run(Thread.java:923)
I/OneSignal( 6899): Restoring notifications`

**Comment by giovanni256** [comment]
@nan-li thanks, see the solution proposed by the RevenueCat Team [here](https://github.com/OneSignal/OneSignal-Android-SDK/issues/1595#issuecomment-1209349671)

**Comment by nan-li** [comment]
Hi all, this is fixed in OneSignal-Flutter-SDK [release 3.4.1](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/3.4.1).
